src/main/java/fr/insalyon/creatis/gasw/executor/batch/internals/BatchManager.java

```diff
@@ -0,0 +1,191 @@
+import fr.insalyon.creatis.gasw.GaswException;
+import fr.insalyon.creatis.gasw.execution.GaswStatus;
+import fr.insalyon.creatis.gasw.executor.batch.config.json.properties.BatchConfig;
+import fr.insalyon.creatis.gasw.executor.batch.internals.commands.RemoteCommand;
+import fr.insalyon.creatis.gasw.executor.batch.internals.commands.items.Mkdir;
+import fr.insalyon.creatis.gasw.executor.batch.internals.commands.items.Rm;
+import fr.insalyon.creatis.gasw.executor.batch.internals.terminal.RemoteFile;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.extern.log4j.Log4j;
+
+@Getter
+@Log4j
+public class BatchManager {
+
+    final private String            workflowId;
+    final private BatchConfig       config;
+    final private String            workingDir;
+    final private List<BatchJob>    jobs = new ArrayList<>();
+
+    private boolean initialized = false;
+    private Boolean end;
+
+    public BatchManager(final String workflowId, final BatchConfig config) {
+        this.workflowId = workflowId;
+        this.config = config;
+        this.workingDir = config.getCredentials().getWorkingDir() + workflowId;
+    }
+
+    public void init() {
+        try {
+            checkRemoteDirs();
+            checkLocalOutputsDir();
+
+            initialized = true;
+        } catch (GaswException e) {
+            log.error("Failed to init the batch manager !");
+        }
+    }
+
+    /**
+     * Create the necessary directories on the remote (batch cluster)
+     */
+    public void checkRemoteDirs() throws GaswException {
+        final List<RemoteCommand> commands = new ArrayList<>();
+
+        commands.add(new Mkdir(getWorkingDir(), ""));
+        commands.add(new Mkdir(getWorkingDir() + "/out", "-p"));
+        commands.add(new Mkdir(getWorkingDir() + "/err", "-p"));
+        commands.add(new Mkdir(getWorkingDir() + "/sh", "-p"));
+
+        for (final RemoteCommand command : commands) {
+            command.execute(config);
+
+            if (command.failed()) {
+                log.error("Failed to create the remotes dirs: " + command.getCommand());
+                throw new GaswException("Failed to create the remotes dirs: " + command.getCommand());
+            }
+        }
+    }
+
+    public void checkLocalOutputsDir() {
+        final String[] dirs = { GaswConstants.OUT_ROOT, GaswConstants.ERR_ROOT, "./cache" };
+
+        Arrays.stream(dirs)
+                .map(File::new)
+                .forEach(dir -> {
+                    if (!dir.exists()) {
+                        dir.mkdirs();
+                    }
+                });
+    }
+
+    /**
+     * Clean files created inside the workflow folder
+     */
+    public void destroy() {
+        final RemoteCommand remoteCommand = new Rm(config.getCredentials().getWorkingDir() + workflowId, "-rf");
+
+        end = true;
+        try {
+            if (remoteCommand.execute(config).failed()) {
+                throw new GaswException("");
```

Comment:
add message and log

Reply:
Actually you catch it just afterward, but not pretty.
So just log if something goes wrong and ignore the errors.